Add spinup diagnostic presets to historical+concentrations configuration

blimlim opened the pull request:
Closes the historical half of #104.

This pull request adds a new "spinup" output preset for each model component. These request a reduced amount of output, tailored to long spinup simulations, producing ~ 900M output per year. 

The changes relative to the "standard" preset are listed in the following spreadsheet:
[spinup_preset (1).xls](https://github.com/user-attachments/files/17970655/spinup_preset.1.xls)

Four new variables were also added to all three presets:
* temp_yflux_adv_int_z
* temp_xflux_adv_int_z
* salt_yflux_adv_int_z
* salt_xflux_adv_int_z

as discussion identified them as useful.

MartinDix commented:
I'd like to keep sub-surface runoff because it's necessary for calculating a water budget.

blimlim commented:
> I'd like to keep sub-surface runoff because it's necessary for calculating a water budget.

Good idea! I've added this back in (8235), and also added back "geopotential height on plev grid" (30207) as suggested on the [Hive Forum](https://forum.access-hive.org.au/t/preset-outputs-for-paleoclimate-simulations/3678/20)

MartinDix reviewed (approved):
Happy with this now
